### PR TITLE
Change package to ServiceContext; keep baggage compat module

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/apple/swift-distributed-tracing-baggage/issues/31

We are going to make a big rename here.

Due to many teams actively using this type in production right now even, and sometimes OUTSIDE of the context of distributed tracing just as a "glorified type-safe typesafe dictionary" we are going to embrace that that's what this type is, and going to call it ServiceContext.

Truth be told, this type wanted to be called this long time ago but it was back then decided that Context was too generic and we picked a tracing only name. Reality has shown us that the type is much more useful and we should not limit its wording and usage to tracing.

Thanks for opening the discussion and I'll do the necessary changes including repo name change.

### CODE REVIEW

**Review by tomerd — APPROVED** [review]

File: Sources/InstrumentationBaggage/Shims.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@exported import ServiceContextModule
+
+public typealias Baggage = ServiceContext
+public typealias BaggageKey = ServiceContextKey
+public typealias AnyBaggageKey = AnyServiceContextKey
```
Comment (stevapple) [inline Sources/InstrumentationBaggage/Shims.swift:20]: `@available(*, deprecated)` can be used with `typealias` to provide warnings, so I think we can provide diagnostic messages here.

```suggestion
@available(*, deprecated, message: "Use 'ServiceContext' from 'ServiceContextModule' instead.")
public typealias Baggage = ServiceContext
@available(*, deprecated, message: "Use 'ServiceContextKey' from 'ServiceContextModule' instead.")
public typealias BaggageKey = ServiceContextKey
@available(*, deprecated, message: "Use 'AnyServiceContextKey' from 'ServiceContextModule' instead.")
public typealias AnyBaggageKey = AnyServiceContextKey
```

Emitting:
```
warning: 'Baggage' is deprecated: Use 'ServiceContext' from 'ServiceContextModule' instead.
```

Comment (ktoso) [reply Sources/InstrumentationBaggage/Shims.swift]: Yes those will become deprecated; I've but this up a bit early as a draft to check CI

**Comment by ktoso** [comment]
And we're green -- and deprecated the Baggage names as well already.

**Comment by ktoso** [comment]
@weissi signed off via chat :+1:

**Comment by tomerd** [comment]
@ktoso do we need to rename the repo?

**Pull request merged**